### PR TITLE
Cast `Uri@filterPort()` `$port` parameter to string if non-null

### PR DESCRIPTION
**Pull request by cosmastech**
To resolve https://github.com/slimphp/Slim-Psr7/issues/288

### CODE REVIEW

File: src/Uri.php [hunk 1]
```diff
@@ -300,7 +300,13 @@ protected function hasStandardPort(): bool
      */
     protected function filterPort($port): ?int
```
Comment (l0gicgate) [inline src/Uri.php:301]: Can you update the doc bloc that it can take a string or an int?

Comment (cosmastech) [reply src/Uri.php:301]: done

**Comment by l0gicgate** [comment]
@cosmastech it appears that CI is failing on 8.2 now

**Comment by cosmastech** [comment]
> @cosmastech it appears that CI is failing on 8.2 now

This isn't related to anything I changed (and is also failing on https://github.com/slimphp/Slim-Psr7/pull/294/files). Maybe should be a separate PR to remove that phpstan rule?

**Comment by akrabat** [comment]
Thanks. I'm sorry that it took so long.

**Comment by cosmastech** [comment]
> Thanks. I'm sorry that it took so long.

No problem. Thanks for letting me contribute!

**Pull request merged**